src/projectile.py

```diff
@@ -22,53 +22,100 @@
     from player import Player
 
 
-class Projectile:
+class Ordnance:
     """
-    A projectile can be anything that is created by a weapon or another projectile included beams, rockets, etc.
-    It stores the characteristics of the projectile and is responible the associated sprite
+    An ordnance can be anything that is created by a weapon or another ordnance included beams, rockets, explosions, etc.
+    Ordnace can be conatined in the payload of another ordance to be later activated under the right conditions
+    Although by definition, the term ordnance includes weapons as well, this class does not contain the weapons themselves
     """
 
-    sprite: LinkedSprite[Projectile]
+    sprite: LinkedSprite[Ordnance]
     sprite_lists: SpriteLists
-    muzzle_location: Tuple[float, float, float]
-    damage: float
-    speed: float
-    angle_of_motion: float
+    payload_list: list[Ordnance]
+    """
+    List of Ordnance that are spawned when this Ordnance collides with something. This could be explosions, hit indicators, more rockets that go in every direction, etc.
+    """
     sprite_rotation_offset: float
     exists: bool
     """
-    Projectile is visible and can be collided with.  Used for long-lived projectile objects that are repeatedly added to/removed from the world over time, such as laser beams.
+    Ordnance is visible and can be collided with.  Used for long-lived projectile objects that are repeatedly added to/removed from the world over time, such as laser beams.
     """
 
-    explodes: bool
-
     def __init__(
         self,
-        sprite: LinkedSprite[Projectile],
+        sprite: LinkedSprite[Ordnance],
         sprite_lists: SpriteLists,
-        damage: float,
+        payload_list: list[Ordnance],
     ):
         self.sprite = sprite
         sprite.owner = self
         self.sprite_lists = sprite_lists
-        self.damage = damage
+        self.payload_list = payload_list
+        self.sprite_rotation_offset = 0
+        self.exists = False
+
+    @property
+    def location(self) -> Tuple[float, float, float]:
+        return (
+            self.sprite.center_x,
+            self.sprite.center_y,
+            self.sprite.radians - self.sprite_rotation_offset,
+        )
+
+    def append_sprite(self):
+        self.sprite_lists.ordnance.append(self.sprite)
+        self.exists = True
+
+    def remove_sprite(self):
+        self.sprite_lists.ordnance.remove(self.sprite)
         self.exists = False
-        self.speed = 0
-        self.angle_of_motion = 0
 
-    def setup(
+    def activate_payload(self):
```

Comment:
I'm not sure this method belongs at this layer of abstraction.  I see it is always being called, even if there is no weapon payload.  I think there might be a better way of implementing the polymorphism here.

Reply:
This is here because most ordnance will end up having a payload eventually once hit markers are implemented. These will be quick small flashes that indicate something got hit